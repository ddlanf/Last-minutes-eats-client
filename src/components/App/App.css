@import url("https://fonts.googleapis.com/css2?family=Permanent+Marker&display=swap");
@import url("https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,300;1,300&display=swap");

.sign-up-form > *, 
.delete-recipe-form > *, 
.landing-page > *, 
.edit-recipe > *, 
.make-recipe > *,
.not-found-page,
.view-all-recipes-recipes > *,
.view-recipe-recipe > *{
    font-family: 'Baloo Tamma 2';
}