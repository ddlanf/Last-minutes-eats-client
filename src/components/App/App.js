import React from 'react';
import './App.css'
import { Route, Switch } from 'react-router-dom'
import Nav from '../Nav/Nav'
import LandingPage from '../LandingPage/LandingPage'
import SignUp from '../SignUp/SignUp'
import NotFoundPage from '../NotFoundPage/NotFoundPage'
import ViewAllRecipes from '../ViewAllRecipes/ViewAllRecipes'
import ViewRecipe from '../ViewRecipe/ViewRecipe'
import EditRecipe from '../EditRecipe/EditRecipe'
import MakeRecipe from '../MakeRecipe/MakeRecipe'

function App() {
  return (
    <div className='App'>
        <header className='App_header'>
            <Nav />
        </header>
        <main className="App_main">
          <Switch>
            <Route
              exact
              path={'/'}
              component={LandingPage}
            />
            <Route
              exact
              path={'/signup'}
              component={SignUp}
            />
             <Route
              exact
              path={'/view-all-recipes'}
              component={ViewAllRecipes}
            />
             <Route
              exact
              path={'/view-recipe/:recipeId'}
              component={ViewRecipe}
            />
            <Route
              exact
              path={'/edit-recipe/:recipeId'}
              component={EditRecipe}
            />
             <Route
              exact
              path={'/make-recipe'}
              component={MakeRecipe}
            />
            <Route
              component={NotFoundPage}
            />
          </Switch>
        </main>
    </div>
  );
}

export default App;