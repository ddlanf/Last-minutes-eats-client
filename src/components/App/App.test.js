import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import rootReducer from '../../reducers/index'

describe('App component', () => {
    
    const store = createStore(rootReducer, window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__());

    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
                      <Provider store={store}>
                        <App/>
                      </Provider>
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(<BrowserRouter>
                    <Provider store={store}>
                      <App/>
                    </Provider>
                  </BrowserRouter>)
          .toJSON();
        expect(tree).toMatchSnapshot();  
      });
});