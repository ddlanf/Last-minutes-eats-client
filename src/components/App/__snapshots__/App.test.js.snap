// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders the UI as expected 1`] = `
<div
  className="App"
>
  <header
    className="App_header"
  >
    <nav
      className="nav-bar"
    >
      <div
        className="nav-bar-left-nav"
      >
        <a
          className="nav-bar-left-nav-link"
          href="/view-all-recipes"
          onClick={[Function]}
        >
          Last Minute Eats
        </a>
      </div>
      <div
        className="nav-bar-right-nav"
      >
        <a
          className="nav-bar-right-nav-link"
          href="/signup"
          onClick={[Function]}
        >
          Sign Up
        </a>
      </div>
    </nav>
    <div
      className="
                    logo 
                    
                "
      onClick={[Function]}
    >
      <p>
        Click me to get started
      </p>
      <img
        alt="logo"
        className="logo-image"
        src="onigiri.png"
      />
    </div>
  </header>
  <main
    className="App_main"
  >
    <div
      className="lading-page-top-sections"
    >
      <section
        className="landing-page-hero"
      >
        <img
          alt="pizza"
          className="food-image"
          onAnimationEnd={[Function]}
          src="pizza.jpg"
          style={
            Object {
              "display": "block",
            }
          }
        />
        <img
          alt="chicken"
          className="food-image"
          onAnimationEnd={[Function]}
          src="chicken.jpg"
          style={
            Object {
              "display": "none",
            }
          }
        />
        <img
          alt="noodle"
          className="food-image"
          onAnimationEnd={[Function]}
          src="noodle.png"
          style={
            Object {
              "display": "none",
            }
          }
        />
        <img
          alt="rice"
          className="food-image"
          onAnimationEnd={[Function]}
          src="rice.jpg"
          style={
            Object {
              "display": "none",
            }
          }
        />
        <img
          alt="taco"
          className="food-image"
          onAnimationEnd={[Function]}
          src="tacos.jpg"
          style={
            Object {
              "display": "none",
            }
          }
        />
      </section>
      <div
        className="lading-page-headings"
      >
        <section
          className="landing-page-heading first"
        >
          <h1
            className="landing-page-heading-text"
          >
            Woke up late to work in the morning or feeling lazy to prepare a whole meal?
            <span
              className="landing-page-break"
            >
              Last minute eats has the perfect recipes for you
            </span>
          </h1>
        </section>
        <section
          className="landing-page-heading second"
        >
          <h1
            className="landing-page-heading-text"
          >
            Your time is valuable.
            <span
              className="landing-page-break"
            >
              That's why all of our recipes take less than 15 minutes to prepare.
            </span>
          </h1>
          <div
            className="landing-page-heading-small-text"
          >
            Average Americnans spend 2 hours a day preparing meals.
            <span
              className="landing-page-break"
            >
              That's 30 days a year spent on cooking alone!
            </span>
            <span
              className="landing-page-break"
            >
              Our recipes has less than 5 ingredient and fewer than 7 short steps to have your stomach filled
            </span>
          </div>
        </section>
      </div>
    </div>
    <section
      className="landing-page-my-info"
    >
      <p
        className="landing-page-my-name"
      >
        Created by Daniel Laufenberg
      </p>
      <div
        className="github-and-linkedIn-links-mobile"
      >
        <a
          href="https://github.com/ddlanf"
          target="_blank"
        >
          <i
            className="fa"
          />
        </a>
        <a
          href="https://www.linkedin.com/in/daniel-laufenberg/"
          target="_blank"
        >
          <i
            className="fa"
          />
        </a>
      </div>
      <div
        className="github-and-linkedIn-links"
      >
        <a
          href="https://github.com/ddlanf"
          target="_blank"
        >
          <i
            className="fa"
          />
        </a>
        <a
          href="https://www.linkedin.com/in/daniel-laufenberg/"
          target="_blank"
        >
          <i
            className="fa"
          />
        </a>
      </div>
      <p
        className="landing-page-copy"
      >
        Copyright Â© 2020
      </p>
      <p
        className="landing-page-rights-reserved"
      >
        All Rights Reserved
      </p>
    </section>
  </main>
</div>
`;
